GenderPayGap.WebUI/Controllers/Admin/AdminOrganisationCompanyNumberController.cs

```diff
@@ -0,0 +1,235 @@
+            return View("OfferChangeOrRemoveOrganisationCompanyNumber", viewModel);
+        }
+
+        [HttpPost("organisation/{id}/change-company-number")]
+        public IActionResult ChangeCompanyNumberPost(long id, AdminOrganisationCompanyNumberViewModel viewModel)
+        {
+            Organisation organisation = dataRepository.Get<Organisation>(id);
+            
+            switch (viewModel.CurrentPage)
+            {
+                case AdminOrganisationCompanyNumberViewModelCurrentPage.OfferChangeOrRemove:
+                    return OfferChangeOrRemove(organisation, viewModel);
+                case AdminOrganisationCompanyNumberViewModelCurrentPage.Remove:
+                    return Remove(organisation, viewModel);
+                case AdminOrganisationCompanyNumberViewModelCurrentPage.Change:
+                    return Change(organisation, viewModel);
+                case AdminOrganisationCompanyNumberViewModelCurrentPage.BackToChange:
+                    return BackToChange(organisation, viewModel);
+                case AdminOrganisationCompanyNumberViewModelCurrentPage.ConfirmNew:
+                    return ConfirmNew(organisation, viewModel);
+                default:
+                    throw new ArgumentOutOfRangeException();
+            }
+        }
+
+        private IActionResult OfferChangeOrRemove(Organisation organisation, AdminOrganisationCompanyNumberViewModel viewModel)
+        {
+            viewModel.Organisation = organisation;
+
+            viewModel.ParseAndValidateParameters(Request, m => m.ChangeOrRemove);
+
+            if (viewModel.HasAnyErrors())
+            {
+                return View("OfferChangeOrRemoveOrganisationCompanyNumber", viewModel);
+            }
+
+            // If removing CompanyNumber
+            if (viewModel.ChangeOrRemove == AdminOrganisationCompanyNumberChangeOrRemove.Remove)
+            {
+                return View("RemoveOrganisationCompanyNumber", viewModel);
+            }
+
+            // If changing CompanyNumber
+            return View("ChangeOrganisationCompanyNumber", viewModel);
+        }
+
+        private IActionResult Remove(Organisation organisation, AdminOrganisationCompanyNumberViewModel viewModel)
+        {
+            viewModel.ParseAndValidateParameters(Request, m => m.Reason);
+
+            if (viewModel.HasAnyErrors())
+            {
+                viewModel.Organisation = organisation;
+                return View("RemoveOrganisationCompanyNumber", viewModel);
+            }
+
+            auditLogger.AuditChangeToOrganisation(
+                AuditedAction.AdminChangeOrganisationCompanyNumber,
+                organisation,
+                new
+                {
+                    OldCompanyNumber = organisation.CompanyNumber,
+                    NewCompanyNumber = "(removed)",
+                    Reason = viewModel.Reason
+                },
+                User

```

Comment:
I just added a small commit to re-add this User into audit logging (it was causing an error when you view the audit logs if the user isn't set).
But, other than that, looks great 😄